runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -164,6 +191,241 @@ private Function<Function<String, String>, String> asAccessor(
+                {
+                    isMatch = parseParams(request, path);
+                }
+                if (isMatch)
+                {
+                    result = request;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private boolean parseParams(
+        HttpRequestType request,
+        String path)
+    {
+        boolean result = false;
+        String pattern =
+            "^" +
+            request.path.replaceAll("\\{([a-zA-Z_]+)\\}", "(?<$1>.+?)") +
+            "/?(\\?(?<query0>.*))?$";
+        Matcher matcher = Pattern.compile(pattern).matcher(path);
+        if (matcher.matches())
+        {
+            this.pathParams = parsePathParams(request.path, matcher);
+            this.queryParams = parseQueryParams(matcher.group("query0"));
+            result = true;
+        }
+        return result;
+    }
+
+    private Map<String, String8FW> parsePathParams(
+        String pathTemplate,
+        Matcher matcher)
+    {
+        Map<String, String8FW> result = new HashMap<>();
+        String[] segments = pathTemplate.split("/");
+        for (String segment : segments)
+        {
+            if (segment.startsWith("{") && segment.endsWith("}"))
+            {
+                String name = segment.substring(1, segment.length() - 1);
+                String8FW value = new String8FW(URLDecoder.decode(matcher.group(name), UTF_8));
+                result.put(name, value);
+            }
+        }
+        return result;
+    }
+
+    private Map<String, String8FW> parseQueryParams(
+        String query)
+    {
+        Map<String, String8FW> queryParams = new HashMap<>();
+        if (query != null && !query.isBlank())
+        {
+            String[] paramPairs = query.split("&");
+            for (String paramPair : paramPairs)
+            {
+                String[] keyValue = paramPair.split("=");
+                if (keyValue.length == 2)
+                {
+                    String key = URLDecoder.decode(keyValue[0], UTF_8);
+                    String8FW value = new String8FW(URLDecoder.decode(keyValue[1], UTF_8));
+                    queryParams.put(key, value);
+                }
+            }
+        }
+        return queryParams;
+    }
+
+    public boolean validateHeader(
+        HttpRequestType request,
+        HttpBeginExFW beginEx)
+    {
+        boolean isValid = true;
+        if (request != null)
+        {
+            boolean isValidHeader = validateHeaderValues(request, beginEx);
+            boolean isValidPathParams = validatePathParams(request);
+            boolean isValidQueryParams = validateQueryParams(request);
+            isValid = isValidHeader && isValidPathParams && isValidQueryParams;
+        }
+        return isValid;
+    }
+
+    private boolean validateHeaderValues(
+        HttpRequestType request,
+        HttpBeginExFW beginEx)
+    {
+        AtomicBoolean isValid = new AtomicBoolean(true);
```

Comment:
Use `MutableBoolean` instead of `AtomicBoolean`.

Reply:
done